library/L1_Peripheral/stm32f10x/pwm.hpp

```diff
@@ -0,0 +1,296 @@
+#pragma once
+
+#include "L0_Platform/stm32f10x/stm32f10x.h"
+#include "L1_Peripheral/pwm.hpp"
+#include "L1_Peripheral/stm32f10x/pin.hpp"
+#include "L1_Peripheral/stm32f10x/system_controller.hpp"
+#include "utility/log.hpp"
+
+namespace sjsu::stm32f10x
+{
+/// An implementation of the sjsu::Pwm peripheral for the stm32f10x family of
+/// microcontrollers.
+class Pwm final : public sjsu::Pwm
+{
+ public:
+  static constexpr auto kOutputChannelEnable = bit::MaskFromRange(0);
+  static constexpr auto kTimerEnable         = bit::MaskFromRange(0);
+  static constexpr auto kUpdateGeneration    = bit::MaskFromRange(0);
+
+  // We use pins spread out across Timers 1 - 5
+  struct PwmPin_t
+  {
+    // Reference to the pin
+    sjsu::Pin & pin;
+
+    // Holds the address to the corresponding Timer peripheral
+    TIM_TypeDef * registers;
+
+    // Holds the channel (1 - 4) which corresponds to the pin
+    uint8_t channel : 3;
```

Comment:
```suggestion
    uint8_t channel;
```
Remove the bit field. It can cause really annoying warning and errors and it doesn't save us any space without making the struct packed. Also don't make the struct packed because you end up wasting space in the generated code to pull out those 3 bits than you would get back form packing this struct.